src/EVEMon/SkillPlanner/PlanWindow.cs

```diff
@@ -777,6 +814,86 @@ internal void UpdateStatusBar()
         }
 
+
+        /// <summary>
+        /// Imports list of skills from the clipboard to the training plan
+        /// </summary>
+        /// <param name="text">Clipboard contents.</param>
+        internal void ImportSkillsFromClipboard(string text)
+        {
+            string[] lines = text.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
+
+            List<StaticSkillLevel> list = new List<StaticSkillLevel>();
+
+            // Nothing to evaluate
+            if (lines.Length == 0)
+                return;
+
+            CharacterScratchpad scratchpad = new CharacterScratchpad(m_character);
+
+            foreach (string line in lines)
+            {  
+
+                // Split level and skill
+                int idx = line.LastIndexOf(" ");
+                if (idx != -1)
+                {
+                    string name = line.Substring(0, idx);
+                    string level = line.Substring(idx + 1);
+                    StaticSkillLevel skill = new StaticSkillLevel(name, Skill.GetIntFromRoman(level));
+
+                    // Make sure we actually have a valid skill
+                    if(skill.Skill != StaticSkill.UnknownStaticSkill)
+                    {
+                        // Add any dependencies that the skill may have
+                        scratchpad.Train(skill.AllDependencies.Where(x => m_character.Skills[x.Skill.ID].Level < x.Level));
+
+                        // Add the skill itself
+                        scratchpad.Train(skill);
+                    }   
+                }
+
+            }
+
+            // Add all trained skills to a list
+            list.AddRange(scratchpad.TrainedSkills);
+
+            if(list.Count == 0)
+            {
+                MessageBox.Show(@"Pasted skills and all dependencies have already been trained.", @"Already Trained",
+                    MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
+            }
+            else if (m_plan.AreSkillsPlanned(list))
+            {
+                MessageBox.Show(@"Pasted skills and all dependencies have already been trained or planned.", @"Already Trained or Planned",
+                                    MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
+            }
+            else
+            {
+                TimeSpan trainingTime = m_character.GetTrainingTimeToMultipleSkills(list);
+                string trainingDesc = trainingTime.ToDescriptiveText(DescriptiveTextOptions.IncludeCommas | DescriptiveTextOptions.SpaceText);
+
+                DialogResult dr = MessageBox.Show($"Are you sure you want to add {list.Count} skills" +
+                                                    $" with a total training time of {trainingDesc}" +
+                                                    $".\n\n This will also include any dependencies not included in your paste", "Add Skills?",
```

Comment:
My OCD really dislikes the space here before "This".
The text in the two sentences isn't aligned because of it.

Reply:
Ha, I think that ended up there as I originally wasn't going to break it over too lines.

I'll fix this one but I'm in 2 minds as to whether the dialog is necessary as you get another one straight after!